File: BussinesLogic/Servicios/ServicioCategorias.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;
using DataAccess.StoredProcedures;
using Dominio.Interfaces.Aplicacion.Categoria;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using Dominio.Dto;
using Dominio.Entiti;

namespace BussinesLogic.Servicios
{
    public class ServicioCategorias : IServicioCategorias
    {

        private readonly IRepositorioCategorias repositorioCategorias;

        public ServicioCategorias(IRepositorioCategorias repositorioCategorias)
        {
            this.repositorioCategorias = repositorioCategorias;
        }
        #region Selects
        public async Task<List<DtoCategorias>> ObtenerCategorias()
        {
            return await this.repositorioCategorias.ObtenerCategorias();
        }
        public async Task<DtoCategorias> ObtenerCategoriasPorId(int idCategoria)
        {
            return await this.repositorioCategorias.ObtenerCategoriasPorID(idCategoria);
        }



        #endregion
        public async Task<DtoRespuestaSP> InsertarCategoria(EntitiCategoria entitiCategoria)
        {
            return await this.repositorioCategorias.InsertarCategoria(entitiCategoria);
        }
        public async Task<DtoRespuestaSP> ActualizarCategoria(EntitiCategoria entitiCategoria)
        {
            return await this.repositorioCategorias.ActualizarCategoria(entitiCategoria);
        }
        public async Task<DtoRespuestaSP> InactivarCategoria(int idCategoria)
        {
            return await this.repositorioCategorias.EliminarCategoria(idCategoria);
        }


    }
}

File: CX/ClientApp/src/servicios/ServicioCategorias.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "categorias";

export const ObtenerCategorias = async () => {
    const url = `${controlador}/ObtenerCategorias`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerCategoriasPorId = async (id) => {
    const url = `${controlador}/ObtenerCategoriasPorId/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const InsertarCategorias = async (Data) => {
    const url = `${controlador}/InsertarCategoria`;
    return await ProcesarDatosApi('POST', url, Data);
}

export const ActualizarCategoria = async (Data) => {
    const url = `${controlador}/ActualizarCategoria`;
    return await ProcesarDatosApi('PUT', url, Data);
}

export const InactivarCategoria = async (id) => {
    const url = `${controlador}/InactivarCategoria?idCategoria=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}


